src/resource.rs

```diff
@@ -565,14 +539,28 @@ impl Resource {
         }
     }
 
+    /// Returns the number of operations of this [`Resource`].
+    pub fn num_operations(&self) -> usize {
+        self.operations.len()
+    }
+
+    /// Returns the number of types of this [`Resource`].
+    pub fn num_types(&self) -> usize {
+        self.types.len()
+    }
     /// Allows read-only access to the operations in this Resource
-    pub fn operations(&self) -> &HashMap<SmolStr, Arc<OpDef>> {
-        &self.operations
+    pub fn get_op(&self, op_name: &str) -> Option<&Arc<OpDef>> {
+        self.operations.get(op_name)
     }
 
     /// Allows read-only access to the types in this Resource
-    pub fn types(&self) -> &HashMap<SmolStr, TypeDef> {
-        &self.types
+    pub fn get_type(&self, type_name: &str) -> Option<&TypeDef> {
+        self.types.get(type_name)
+    }
+
+    /// Allows read-only access to the operations in this Resource
+    pub fn get_op_mut(&mut self, op_name: &str) -> Option<&mut Arc<OpDef>> {
```

Comment:
Is this method useful?

Reply:
Used by `OpDef::add_lowering`

Reply:
Hmm, can't find `get_op_mut` used anywhere on this branch at the moment?

Reply:
sorry I mean, in order for `add_lowering` to be used, you would need a mutable reference. I guess I can add a test for that function.

Reply:
Ah, right I see!